### PR TITLE
Fixes #554 representation of new model instances

### LINKED ISSUE #554: representation/printing of Models fails when model is not yet persisted to the database
When you create an new instance of a model you cannot print it or use it in logs, because the __repr__ method of Model tries to loop through the identity parameter of inspect(), which is None. 

After persisting to the database the instance can be printed.

example: 
```python
>>> from application.extensions import db
>>> from application.model.models import User
>>> u = User()
>>> print(u)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/jeff/virtualEnvs/kongoserver/lib/python3.5/site-packages/flask_sqlalchemy/model.py", line 121, in __repr__
    pk = ', '.join(to_str(value) for value in inspect(self).identity)
TypeError: 'NoneType' object is not iterable
>>> u.id = 12345
>>> print(u)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/jeff/virtualEnvs/kongoserver/lib/python3.5/site-packages/flask_sqlalchemy/model.py", line 121, in __repr__
    pk = ', '.join(to_str(value) for value in inspect(self).identity)
TypeError: 'NoneType' object is not iterable
>>> db.session.add(u)
>>> db.session.commit()
>>> print(u)
<User 12345>
```

Where User is defined like: 
```python
class User(db.Model):
    __tablename__ = 'application_user'

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(64), unique=True)
    password = db.Column(db.String(255))
```

### PR DESCRIPTION
**Pull request by jeff-99**
This fixes issue #554 by printing an id of "None" before a model is persited to the database

### CODE REVIEW

**Comment by davidism** [comment]
It would be nice to show something about each instance still, for example `<Model (transient) id(self)>`.

**Comment by jeff-99** [comment]
Great idea, I've added the 'transient' id to the __repr__ method

**Comment by davidism** [comment]
Needs a test.

Let's make it `(transient id(self))` instead so it's a little clearer the id is part of the transient mark instead of the actual key.

File: tests/test_model_class.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ class Report(db.Model):
     db.create_all()
 
     u = User(name='test')
+    assert repr(u).startswith("<User (transient ") is True
```
Comment (davidism) [inline tests/test_model_class.py]: remove `is True`, it's redundant

**Comment by davidism** [comment]
Looks like that latest commit was misconfigured, your git name and email don't match GitHub.

**Comment by davidism** [comment]
Oops, forgot to rebase this to 2.3-maintenance. I'll cherry pick it.

**Pull request merged**